hive/server/condenser_api/common.py

```diff
@@ -1,37 +1,64 @@
 """Helpers for condenser_api calls."""
 
 import re
+from functools import wraps
 
 from hive.db.methods import query_one, query_col
 
+class ApiError(Exception):
+    """API-specific errors: unimplemented/bad params. Pass back to client."""
+    pass
+
+def return_error_info(function):
+    """Async API method decorator which catches and formats exceptions."""
+    @wraps(function)
+    async def wrapper(*args, **kwargs):
+        """Catch ApiError and AssersionError (always due to user error)."""
+        try:
+            return await function(*args, **kwargs)
+        except (ApiError, AssertionError) as e:
+            return {
+                "error": {
+                    "code": -32000,
+                    "message": str(e) + " (hivemind-alpha)"}}
+    return wrapper
+
 def valid_account(name, allow_empty=False):
     """Returns validated account name or throws Assert."""
+    if not name:
+        assert allow_empty, 'account must be specified'
+        return ""
     assert isinstance(name, str), "account must be string; received: %s" % name
-    if not (allow_empty and name == ''):
-        assert len(name) >= 3 and len(name) <= 16, "invalid account: %s" % name
-        assert re.match(r'^[a-z0-9-\.]+$', name), 'invalid account char'
+    assert len(name) >= 3 and len(name) <= 16, "invalid account: %s" % name
```

Comment:
just a question.  on `assert` usually i've seen that for debug tasks.  this looks like we're using it as validators.  not a critique, more confirmation.

Reply:
The sync/indexer side of things uses `assert`s judiciously for sanity-checking (and visibly establishing) any core assumptions to ensure things blow up immediately and loudly. On the API side they're functionally equivalent to `if not <assertion>: raise AssertionError("message")` except more readable.